Delta Course (#256)
mitre/saf-training: src/courses/delta/02.md

---
order: 02
next: 03.md
title: 2. Delta Process Requirements
author: Daniel Medina, George Dias
---

## 2.1 Requirements

To conduct the delta process, the following items are needed:

1. An InSpec profile:
    a. Clone or download an existing profile from a repository source to your local environment.
    b. See [What is an InSpec Profile](../beginner/02.md#what-is-an-inspec-profile) for additional information.
2. A guidance file (XCCDF):

>     a. Download either a DISA STIG or a CIS Benchmark from the appropriate site.

Reviewer: Just checking, delta can also handle CIS?

> ::: note Using Generated Profile Stubs

Reviewer: ```suggestion
::: details What if there is no existing InSpec profile?
```

I am assuming you are suggesting this information for an update like Windows 2016 -> Windows 2022 if, for example Windows 2022 did not yet have an InSpec profile? If my assumption is correct, I recommend putting it in a drop down (::: details) rather than (::: note) so people only have to expand and think about it if it is relevant to them.

Author: Agreed. Also framing it as a question the reader would have makes sense.

> Although not a requirement, it is recommended to use a generated profile stub from the `saf generate inspec_profile` command to create a skeleton profile structure based on the XCCDF benchmark guidance (STIG or CIS). This ensures consistency across profiles.

Reviewer: Just checking, delta can also handle CIS?

> Although not a requirement, it is recommended to use a generated profile stub from the `saf generate inspec_profile` command to create a skeleton profile structure based on the XCCDF benchmark guidance (STIG or CIS). This ensures consistency across profiles.

Reviewer: ```suggestion
If the InSpec profile for your target of interest does not exist, you can generated the profile stub from the `saf generate inspec_profile` command to create a skeleton profile structure based on the XCCDF benchmark guidance (STIG or CIS). This ensures consistency across profiles.
```

For more information on this command, refer to the documentation for [`inspec_profile`](https://saf-cli.mitre.org/#inspec-profile).

This command is similar to the Chef InSpec [Initialize a new profile](../beginner/03.md#creating-the-profile), but it uses the XCCDF content to configure (generate) the profile controls metadata and [supporting structure](../beginner/02.md#the-structure-of-an-inspec-profile).
:::

> The [Delta CLI Tools section](./03.md#31-cli-tools) covers specific requirements for both the `delta` and `update_controls4delta` commands.

Reviewer: ```suggestion
```
